server/priv/marketing/blog/2025/08/05/tuist-ai-whitepaper.md

```diff
@@ -0,0 +1,107 @@
+
+It quickly became apparent that these pain points transfer to agents in agentic coding experiences. Agents require rapid feedback to ensure their work progresses in the right direction. When incremental builds work unreliably, causing unexpectedly long compilation times, the experience degrades. Similarly, if previewing changes requires building the app, installing it in a simulator, and launching it, the slow feedback loop compromises the entire experience.
+
+Agents need fast feedback from multiple sources: the build system, running applications, live previews, and more. Some information can be collected because it exists in the system and either the model has learned about it or accesses it through [Model Context Protocols (MCPs)](https://modelcontextprotocol.io/docs/getting-started/intro). However, agents often lack access to crucial information, such as SwiftUI preview trees resolved at runtime or how the build system processes graphs to generate artifacts.
+
+Some improvements remain outside our direct control, such as enhancing framework documentation, minimizing major Swift changes, or Apple decoupling Xcode’s layers and providing programmatic APIs for extension. Whether these changes occur likely depends on Apple’s willingness to trust their developer community beyond just app creation to include tool enhancement.
+
+The community is actively bridging the gap between agents and official tools. This includes [MCPs that expose CRUD APIs for Xcode projects](https://github.com/giginet/xcodeproj-mcp-server) and solutions that approximate hot-reloading workflows by managing xcodebuild and simulator tools. Until underlying tools provide better APIs and documentation, these translations remain essential.
+
+This brings us to our role. As mentioned earlier, shortening feedback loops is crucial, and we believe this should be our primary contribution. Today, this means binary caching based on project generation, but in the future, when caching is built into the build system, we’re working toward creating the fastest, lowest-latency cache solution that agents can access regardless of their location.
+
+Think of it as edge Xcode build cache. We aim to become the infrastructure enabling agents (and developers) to build apps quickly with Xcode. Beyond speed, we’ll automatically collect diagnostic data from underlying tools to help you understand what went wrong or didn’t proceed as planned. This is why we made [ClickHouse](https://clickhouse.com/) a core infrastructure requirement.
+
+## Work done, it’s time to test it
+
+We believe that in coming years, most coding work will be performed by agents with human guidance and additional context sources. As part of this workflow, you’ll want to verify that work has been completed correctly. Traditionally, this involved running tests in CI pipelines, where feature merging depended on human code approval and CI confirmation of passing tests.
```

Comment:
> We believe that in coming years, most coding work will be performed by agents with human guidance and additional context sources.

this is a very bold claim that, sure, lots of people in the industry are making ... but do we actually believe that? I do believe AI and agents will be playing an increasingly important role, but whether that will translate in AI writing most of the code in large codebases is a different story. I would refrain from claiming that coding will be fully done by AI with just a human-in-the-loop.